workos/resources/list.py

```diff
@@ -0,0 +1,88 @@
+from workos.resources.base import WorkOSBaseResource
+
+
+class WorkOSListResource(WorkOSBaseResource):
+    """Representation of a WorkOS List Resource as returned through the API.
+
+    Attributes:
+        OBJECT_FIELDS (list): List of fields a WorkOSListResource is comprised of.
+    """
+
+    OBJECT_FIELDS = ["data", "list_metadata", "metadata"]
+
+    @classmethod
+    def construct_from_response(cls, response):
+        """Returns an instance of WorkOSListResource.
+
+        Args:
+            response (dict): Resource data from a WorkOS API response
+
+        Returns:
+            WorkOSListResource: Instance of a WorkOSListResource with OBJECT_FIELDS fields set
+        """
+        obj = cls()
+        for field in cls.OBJECT_FIELDS:
+            setattr(obj, field, response.get(field))
+
+        return obj
+
+    def to_dict(self):
+        """Returns a dict representation of the WorkOSListResource.
+
+        Returns:
+            dict: A dict representation of the WorkOSListResource
+        """
+        obj_dict = {}
+        for field in self.OBJECT_FIELDS:
+            obj_dict[field] = getattr(self, field, None)
+
+        return obj_dict
+
+    def auto_paginate(self):
```

Comment:
This is looking great!

The only question I have is that I think we discussed making this an `auto_paging_iter` function, rather than a function that loads all the resources.

In the case of an `auto_paging_iter` function, it could be used to iterate through whatever list of objects is returned by the list query, so it's more versatile.

It would also only load resources when iterating through, which may help from overloading the API with many requests at once.

WDYT?

Reply:
This is the reference method from the Stripe SDK I'm thinking of (obviously not a 1:1 comparison): https://github.com/stripe/stripe-python/blob/master/stripe/api_resources/list_object.py#L80

Reply:
I may have misinterpreted what you were talking about when we were chatting about making it an iterator function, apologies if that's the case! As I understand it an iterator function is one that generates a sequence of values on the fly, allowing you to iterate over a large dataset or process a stream of data one item at a time without having to load the entire dataset into memory.

In the last commit, I implemented it incorrectly since it was still building the full list and returning that list as a generator object, which was not correct as it still stores the full list in memory. With this commit, the generator object is created and each page is returned/yielded to the object as it iterates through each page without needing to store the entire list in memory. The method isn't structured exactly like the one from the stripe SDK, but accomplishes the same thing. 

With the method type passed in as part of self now in the metadata, the function is agnostic of the type of list, so it addresses the versatility goal in that way. 

Let me know if I'm off base or want to chat about it further!

Reply:
I can refactor it a bit so that auto_paging_iter() can be called directly on the resource like in your example, but it will be a breaking change. Right now for all Python SDK methods, we are returning a dict and not the resource itself. To call the method directly like this, we'll need to adjust the return of the list methods to return the resource and not the resource as a dictionary.

Do we want to have a breaking change and do a major version bump as part of this PR? With the primary_email method addition we had decided against structuring it in this way to avoid a breaking change.

Reply:
Good point. I have an idea - what if we don't alter the current list methods on the objects (so keeps them backwards compatible), but just add new list methods that can invoke the `auto_paging_iter` and use the new list object. So this would look like:

```
organizations = Organizations.list
for organization in organizations.auto_paging_iter():
  // do something with the orgs
```

Then we'd keep backwards compatibility, we can set the old list methods as deprecated, and the next time we release a major version, we could potentially remove them.

The only issue might be the naming for connections because they are in the SSO object.

Reply:
I think that'll work nicely! I'll circle back to this likely on Monday. 👍🏽